pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -431,7 +403,7 @@ func (e *TokensController) ensureReferencedToken(serviceAccount *v1.ServiceAccou
 	}
 	// Manually add the new token to the cache store.
 	// This prevents the service account update (below) triggering another token creation, if the referenced token couldn't be found in the store
-	e.secrets.Add(createdToken)
+	e.secretLister.Indexer.Add(createdToken)
```

Comment:
> @liggitt Could we now drop this prevention or should we sync/wait?

no, dropping the prevention exposes the controller to racing caches between the two object types. there's not a great condition to wait on...